request_router.h

```diff
@@ -1,21 +1,55 @@
 #ifndef REQUEST_ROUTER_H
 #define REQUEST_ROUTER_H
 
+#include "request_handlers.h"
 #include "server_config.h"
-#include <cstddef>
 
+#include <memory>
+#include <map>
+
+// Stores and allows access to the mapping of (uri_prefix -> RequestHandler),
+// which we call a "route".
+//
+// Also initializes and stores our long-lived RequestHandler instances
 class RequestRouter {
   public:
-    // TODO: Reduce duplication between this and the handlers
+    // Given a ServerConfig, initialize RequestHandler's and map URI prefixes
+    // to them. Prefixes are assumed to be unique within the config file
+    //
+    // Returns true if all routes are built successfully and false otherwise
+    // (e.g., a non-existent handler is named in the config file and fails to be built)
+    //
+    // Example: path /static1 StaticHandler { ... }
+    //          path /static1/subaction DynamicServeHandler { ... }
+    //
+    // would create the mapping:
+    //          /static1 -> StaticHandler*
+    //          /static1/subaction -> DynamicServeHandler*
+    //
+    // Notice the lack of trailing slash: /prefix, not /prefix/
+    bool BuildRoutes(const ServerConfig& server_config);
+
+    // Map full URIs to the appropriate RequestHandler,
+    // with longest-prefix matching for when URIs overlap.
+    // Returns 404 handler when no URI prefix matches
```

Comment:
Since the common API gives a `default` handler in the config file, should we not assume that a 404 handler is used when no URI prefix matches?